lib/mix/lib/mix/scm/git.ex

```diff
@@ -249,25 +258,30 @@ defmodule Mix.SCM.Git do
     end
   end
 
+  @sha1_size 40
+
   defp validate_depth(opts) do
-    case Keyword.take(opts, [:depth, :ref]) do
-      [_, _] ->
-        Mix.raise(
-          "Cannot specify :depth and :ref at the same time. " <>
-            "Error on Git dependency: #{redact_uri(opts[:git])}"
-        )
+    case Keyword.take(opts, [:depth]) do
```

Comment:
We should use Keyword.fetch if we are only looking for a single key.

Reply:
Well spotted!

Reply:
Aha, if we use `Keyword.fetch`, I think we lose the ability to raise if the option is specified more than once?

The current case for `invalid_depth` raises `"The depth must be a positive integer, and be specified only once..."`.

Reply:
Ah, looks good to me then!